### DISCUSSION TITLE
Authentication/authorization for uploaded files

### DISCUSSION THREAD

**Discussion by andrenarchy** [saleor/saleor#12693] [Q&A]
I have an app that generates an invoice for an order, uploads it with the `fileUpload` mutation, and then attaches an invoice with the URL of the uploaded file to the order. It seems like the file is accessible without any authentication when the url is known. Is this intentional? Are there any ways to only make a file accessible to authorized users?

**Comment by timuric** [comment]
Hi @andrenarchy invoices contain hard to guess order token so that anonymous customers could also download files. If you prefer having authentication, you would have to add middleware using something like cloudflare workers
